src/Microsoft.AspNet.OData.Shared/DeltaSetOfT.cs

```diff
@@ -0,0 +1,357 @@
+        /// </summary>
+        public DeltaSet()            
+        {            
+            _clrType = typeof(TStructuralType);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DeltaSet{TStructuralType}"/> class.
+        /// </summary>
+        /// <param name="keys">List of key names for the type</param>
+        public DeltaSet(IList<string> keys)           
+        {
+            _keys = keys;            
+            _clrType = typeof(TStructuralType);            
+        }
+
+    
+       /// <inheritdoc/>      
+        protected override void InsertItem(int index, IDeltaSetItem item)
+        {
+            Delta<TStructuralType> deltaItem = item as Delta<TStructuralType>;
+
+            if (deltaItem == null)
+            {
+                throw Error.Argument("item", SRResources.ChangedObjectTypeMismatch, item.GetType(), typeof(TStructuralType));
+            }
+
+            base.InsertItem(index, item);
+        }
+
+        /// <summary>
+        /// Handler for users Create, Get and Delete Methods
+        /// </summary>
+        internal PatchMethodHandler<TStructuralType> PatchHandler { get; set; }
+
+
+        /// <summary>
+        /// Patch for DeltaSet, a collection for Delta<typeparamref name="TStructuralType"/>
+        /// </summary>
+        /// <param name="originalCollection">Original collection of the Type which needs to be updated</param>
+        /// /// <returns>DeltaSet response</returns>
+        public DeltaSet<TStructuralType> Patch(ICollection<TStructuralType> originalCollection)
+        {
+            PatchHandler = new DefaultPatchHandler<TStructuralType>(originalCollection);
+
+            return CopyChangedValues();
+        }
+
+        /// <summary>
+        /// Patch for DeltaSet, a collection for Delta<typeparamref name="TStructuralType"/>
+        /// </summary>     
+        /// <returns>DeltaSet response</returns>
+        public DeltaSet<TStructuralType> Patch(IPatchMethodHandler patchHandler)
+        {
+            this.PatchHandler = patchHandler as PatchMethodHandler<TStructuralType>;
+            return CopyChangedValues();
+        }
+
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        internal DeltaSet<TStructuralType> CopyChangedValues()
+        {
+            //Here we are getting the keys and using the keys to find the original object 
+            //to patch from the list of collection
+
+            DeltaSet<TStructuralType> deltaSet = CreateDetlaSet();
+
+            foreach (Delta<TStructuralType> changedObj in Items)
+            {
+                DataModificationOperationKind operation = DataModificationOperationKind.Update;
+
+                //Get filtered item based on keys
+                TStructuralType original = null;
+                string errorMessage = string.Empty;
+                string geterrorMessage = string.Empty;
```

Comment:
>geterrorMessage [](start = 23, length = 15)

do you need both errorMessage and geterrorMessage? It looks like you only ever use one or the other...

Reply:
to distinguish gets err message and others(delete/insert)